src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/repository/CaseEventMessageRepository.java

```diff
@@ -54,6 +54,24 @@
             "SELECT * from public.wa_case_event_messages msg where msg.state = 'NEW' "
             + "order by sequence DESC for update skip locked";
 
+    String SELECT_UNPROCESSABLE_MESSAGES =
+            "SELECT message_id from public.wa_case_event_messages msg where msg.state = 'UNPROCESSABLE' "
+            + "for update skip locked";
+
+    String SELECT_READY_MESSAGES =
+            "SELECT message_id from public.wa_case_event_messages msg where msg.state = 'READY' "
+            + "and msg.event_timestamp > msg.event_timestamp + interval '60 minutes' "
+            + "for update skip locked";
+
+    String FIND_PROBLEMATIC_MESSAGES = "SELECT message_id \n"
+                                       + "from public.wa_case_event_messages msg \n"
+                                       + "where msg.state = cast(:#{#messageState.toString()} as message_state_enum) \n"
+                                       + "and case when msg.state='READY' "
+                                       + "then ((current_timestamp - interval '60 minutes') - msg.event_timestamp ) "
+                                       + "> interval '60 minutes' "
+                                       + "else 1=1 end\n"
+                                       + "for update skip locked";
+
```

Comment:
Do you need to lock the records here in all these three queries? As you are reading the records not for updating and not handling the transcation?

Reply:
I don't need to lock the record, will remove it,